resources/aws_ec2.go

```diff
@@ -798,6 +798,13 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 			}
 			_, err := obj.client.StartInstances(startInput)
 			if err != nil {
+				// If the instance is not in a state where it
+				// can be started, we can't do anything.
+				if aerr, ok := err.(awserr.Error); ok {
+					if aerr.Code() == "IncorrectInstanceState" {
```

Comment:
If we're going to have all these absurd strings everywhere let's at least store them as `const` Maybe something like `AwsErrWhatever`...